src/mol-model/structure/structure/util/superposition-sifts-mapping.ts

```diff
@@ -57,19 +58,24 @@ export function alignAndSuperposeWithSIFTSMapping(structures: Structure[]): Alig
     return { entries, zeroOverlapPairs, failedPairs };
 }
 
-function getPositionTables(index: IndexEntry[], pivot: number, other: number, N: number) {
+function getPositionTables(index: IndexEntry[], pivot: number, other: number, N: number, traceOnly: boolean) {
     const xs = MinimizeRmsd.Positions.empty(N);
     const ys = MinimizeRmsd.Positions.empty(N);
 
     let o = 0;
     for (const { pivots } of index) {
-        const a = pivots[pivot];
-        const b = pivots[other];
+        let a = pivots[pivot];
+        let b = pivots[other];
         if (!a || !b) continue;
 
+        if (traceOnly) {
+            a = findTraceAtom(...a);
+            b = findTraceAtom(...b);
+            if (!a || !b) continue;
+        }
```

Comment:
Sorry to be so annoying about this, but I think you should move the trace atom identification to the ``buildIndex`` function. E.g., just make ``start`` index of the trace atom https://github.com/molstar/molstar/blob/9841c773cb2e0dfe716611ad1052f5216903235a/src/mol-model/structure/structure/util/superposition-sifts-mapping.ts#L186 and make ``end = start + 1``. Then you won't need the ``slice`` at the end here and won't have to do ``...a`` in the function call.

Reply:
Good call, and no worries :) I'll get back to this next week.

Reply:
Implemented in [2a00248](https://github.com/molstar/molstar/pull/373/commits/2a002488124d5383eb01a36174ff96019692fae9).

Looks to me as if the chain/residue iterators are still needed, or is it possible to avoid that when in trace-only mode?

Reply:
It's fine to keep them.